README.md

```diff
@@ -537,6 +537,30 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
+* <a id='destructure-inline-let'></a> (<a href='#destructure-inline-let'>link</a>) **When destructuring an enum case or a tuple, place the `let` keyword inline, adjacent to each individual property assignment.** [![SwiftFormat: hoistPatternLet](https://img.shields.io/badge/SwiftFormat-hoistPatternLet-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#hoistPatternLet)
```

Comment:
Putting here so we have a thread:

> 1. This makes it more clear which identifiers are part of the conditional (if / case) check and which identifiers are binding new variables.

I am open to this idea though it would be easier to visualize. What do you think about adding a code example where we bind new variables to show how the proposed style is superior?

Reply:
My main point here is that, with inline let, the `let` keyword is always adjacent to the variables's identifier (`let x`) rather than being off somewhere else in the line. Off the top of my head, this style of enum destructuring is the only case in Swift where the `let` doesn't appear adjacent to the variable.

I can add a code example, like:

```swift
// `let` is adjacent to the variable identifier
case .enumCaseWithSingleAssociatedValue(let string):
case .enumCaseWithMultipleAssociatedValues(let string, let int):

// The `let` keyword is quite far 
case let .enumCaseWithSingleAssociatedValue(string):
case let .enumCaseWithMultipleAssociatedValues(string, int):
```

Reply:
Would this rule apply to `guard` as well as `switch`? I didn't realize we could use the inline `let` syntax with `guard` but guess we can.
![Screen Shot 2021-05-13 at 6 06 47 PM](https://user-images.githubusercontent.com/1791049/118205775-fdc82f80-b415-11eb-9d33-f05c6db16dc6.png)

Reply:
> Would this rule apply to guard as well as switch? 

Yes, this would apply to everywhere where you can pattern match and destructure enums (including `if`, `guard`, etc). I should add a non-`switch` example to the rule.

Reply:
Updated the rule and PR description.